### PR TITLE
userRoute

### PR DESCRIPTION
**Pull request by dkracheli**

### CODE REVIEW

File: controllers/usersController.js [hunk 1]
```diff
@@ -1,3 +1,38 @@
...
+  } catch (err) {
+    return res.status(401).json({ message: err.message });
+  }
+}
+
+async function handleDeleteUser(req, res) {
+  try {
+    await User.find({ email: req.body.email });
+    await oneTimePass.delete({ email: req.body.email });
+    return res.status(200).json({ message: 'The user has been deleted' });
```
Comment (dkracheli) [inline controllers/usersController.js:30]: If the user is with a free plan, you should also delete the account

File: controllers/usersController.js [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+async function handleGetUser(req, res) {
+  const user = await User.retrieve({ email: req.params.email });
+  return res.send(JSON.stringify(user));
+}
+
+async function handleUpdateUser(req, res) {
+  try {
+    console.log(req.body);
+    await User.update({ email: req.body.email }, { data: req.body });
+    return res.status(200).json({ message: 'user update' });
```
Comment (dkracheli) [inline controllers/usersController.js:20]: Missing role check

File: controllers/usersController.js [hunk 3]
```diff
@@ -1,3 +1,38 @@
+const { User } = require('../services/authService');
+const { oneTimePass } = require('../services/registerService');
+
 const getUsers = async (req, res) => res.sendStatus(200);
 
-module.exports = { getUsers };
+async function handleGetUsers(req, res) {
+  const showAllUser = await User.find({});
+  return res.send(JSON.stringify(showAllUser));
```
Comment (dkracheli) [inline controllers/usersController.js:8]: showAllUser can be replaced by "users"

**Review by dkracheli — COMMENTED** [review]
Nice job!!
left some comments

File: controllers/usersController.js (hunk 1 above)
Comment (DaveAvigad) [reply controllers/usersController.js:30]: 1. why delete the account? 
2. why not just set the account as `disabled`/ `closed`?

File: controllers/usersController.js (hunk 3 above)
Comment (DaveAvigad) [reply controllers/usersController.js:8]: agree

File: controllers/usersController.js [hunk 4]
```diff
@@ -1,3 +1,38 @@
...
+}
+
+async function handleGetUser(req, res) {
+  const user = await User.retrieve({ email: req.params.email });
+  return res.send(JSON.stringify(user));
+}
+
+async function handleUpdateUser(req, res) {
+  try {
+    console.log(req.body);
```
Comment (DaveAvigad) [inline controllers/usersController.js:18]: do not forget to remove the redundant `console.log` before the production release.

File: controllers/usersController.js [hunk 5]
```diff
@@ -1,3 +1,38 @@
...
+    await User.update({ email: req.body.email }, { data: req.body });
```
Comment (DaveAvigad) [inline controllers/usersController.js:19]: seems that you push every payload that comes inside the `req.body` make sure you verify what's in-it before 😉

File: index.js [hunk 6]
```diff
@@ -25,8 +25,11 @@ app.use(passport.initialize());
...
 
 app.use('/auth', validation, auth.authRouter);
-app.use('/users', authenticateToken, users.usersRouter);
+app.use('/users', users.usersRouter);
+// authenticateToken
 
-app.all('/', (req, res) => { res.sendFile(path.join(__dirname, './clientPublic/POC.html')); }); // res.redirect('homePage.html')
+app.all('/', (req, res) => {
+  res.sendFile(path.join(__dirname, './clientPublic/POC.html'));
+}); // res.redirect('homePage.html')
```
Comment (DaveAvigad) [inline index.js:41]: do not forget remove commented-out lines before commit

File: routes/usersRoute.js [hunk 7]
```diff
@@ -5,4 +5,10 @@ const usersController = require('../controllers/usersController');
 
 usersRouter.post('/', usersController.getUsers);
 
+usersRouter.get('/', usersController.handleGetUsers);
+usersRouter.get('/:email', usersController.handleGetUser);
+
+usersRouter.put('/:email', usersController.handleUpdateUser);
```
Comment (DaveAvigad) [inline routes/usersRoute.js:12]: 🎉

**Review by DaveAvigad — COMMENTED** [review]
looks good, left some comments :)

**Pull request merged**